In [2]:
# Run this cell before continuing.
library(tidyverse)
library(repr)
library(infer)
library(cowplot)
library(broom)
library(faux)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘cowplot’


The following object is masked from ‘package:lubridate’:

    stamp



************
Welcome to faux. For support and examples visit:
https://debruine.github.io/faux/
- Get and set global package options with: faux_options()
************



# Data

In [3]:
github_data <- read_csv("repositories.csv")
head(github_data)

github_crucial <- github_data[c("Forks","Stars","Watchers")]
tibble(head(github_crucial))

Rows: 215029 Columns: 24
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (8): Name, Description, URL, Homepage, Language, License, Topics, Defau...
dbl  (5): Size, Stars, Forks, Issues, Watchers
lgl  (9): Has Issues, Has Projects, Has Downloads, Has Wiki, Has Pages, Has ...
dttm (2): Created At, Updated At

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Name                       
1 freeCodeCamp               
2 free-programming-books     
3 awesome                    
4 996.ICU                    
5 coding-interview-university
6 public-apis                
  Description                                                                             
1 freeCodeCamp.org's open-source codebase and curriculum. Learn to code for free.         
2 :books: Freely available programming books                                              
3 😎 Awesome lists about all kinds of interesting topics                                  
4 Repo for counting stars and contributing. Press F to pay respect to glorious developers.
5 A complete computer science study plan to become a software engineer.                   
6 A collective list of free APIs                                                          
  URL                                                       Created At         
1 https://github.com/freeCodeCamp/freeCodeCamp              2014-12-24 17:49:19
2 https://github.com/EbookFoundation/free-programming-books 2013-10-11 06:50:37
3 https://github.com/sindresorhus/awesome                   2014-07-11 13:42:37
4 https://github.com/996icu/996.ICU                         2019-03-26 07:31:14
5 https://github.com/jwasham/coding-interview-university    2016-06-06 02:34:12
6 https://github.com/public-apis/public-apis                2016-03-20 23:49:42
  Updated At          Homepage                                                 
1 2023-09-21 11:32:33 http://contribute.freecodecamp.org/                      
2 2023-09-21 11:09:25 https://ebookfoundation.github.io/free-programming-books/
3 2023-09-21 11:18:22 NA                                                       
4 2023-09-21 08:09:01 https://996.icu                                          
5 2023-09-21 10:54:48 NA                                                       
6 2023-09-21 11:22:06 http://public-apis.org                                   
  Size   Stars  Forks Issues ⋯ Has Issues Has Projects Has Downloads Has Wiki
1 387451 374074 33599   248  ⋯  TRUE       TRUE        TRUE          FALSE   
2  17087 298393 57194    46  ⋯  TRUE      FALSE        TRUE          FALSE   
3   1441 269997 26485    61  ⋯  TRUE      FALSE        TRUE          FALSE   
4 187799 267901 21497 16712  ⋯ FALSE      FALSE        TRUE          FALSE   
5  20998 265161 69434    56  ⋯  TRUE      FALSE        TRUE          FALSE   
6   5088 256615 29254   191  ⋯  TRUE      FALSE        TRUE          FALSE   
  Has Pages Has Discussions Is Fork Is Archived Is Template Default Branch
1  TRUE     FALSE           FALSE   FALSE       FALSE       main          
2  TRUE     FALSE           FALSE   FALSE       FALSE       main          
3  TRUE     FALSE           FALSE   FALSE       FALSE       main          
4 FALSE     FALSE           FALSE    TRUE       FALSE       master        
5 FALSE     FALSE           FALSE   FALSE       FALSE       main          
6 FALSE     FALSE           FALSE   FALSE       FALSE       master

Forks Stars  Watchers
1 33599 374074 374074  
2 57194 298393 298393  
3 26485 269997 269997  
4 21497 267901 267901  
5 69434 265161 265161  
6 29254 256615 256615

### Dataset description

The dataset was collected using github search api. The data features 215029 observations and 24 variables of a repository such as the name, description, number of forks, the programming language used, size of the repository, watching number, etc. There are a few attributes that can be worth looking at more closely, including Stars (number of likes the repo has received from users), Forks (number of times the repo has been forked by other users) and Watchers (The number of GitHub users who are "watching" the repository).

### Variables description

* Name: The name of the GitHub repository

* Description: A brief textual description that summarizes the purpose or focus of the repository

* URL: The URL or web address that links to the GitHub repository, which is a unique identifier for the repository

* Created At: The date and time when the repository was initially created on GitHub, in ISO 8601 format

* Updated At: The date and time of the most recent update or modification to the repository, in ISO 8601 format

* Homepage: The URL to the homepage or landing page associated with the repository, providing additional information or resources

* Size: The size of the repository in bytes, indicating the total storage space used by the repository's files and data

* Stars: The number of stars or likes that the repository has received from other GitHub users, indicating its popularity or interest

* Forks: The number of times the repository has been forked by other GitHub users

* Issues: The total number of open issues (indicating bugs, feature requests, or discussions)

* Watchers: The number of GitHub users who are "watching" or monitoring the repository for updates and changes

* Language: The primary programming language

* License: Information about the software license using a license identifier

* Topics: A list of topics or tags associated with the repository, helping users discover related projects and topics of interest

* Has Issues: A boolean value indicating whether the repository has an issue tracker enabled. In this case, it's true, meaning it has an issue tracker

* Has Projects: A boolean value indicating whether the repository uses GitHub Projects to manage and organize tasks and work items

* Has Downloads: A boolean value indicating whether the repository offers downloadable files or assets to users

* Has Wiki: A boolean value indicating whether the repository has an associated wiki with additional documentation and information

* Has Pages: A boolean value indicating whether the repository has GitHub Pages enabled, allowing the creation of a website associated with the repository

* Has Discussions: A boolean value indicating whether the repository has GitHub Discussions enabled, allowing community discussions and collaboration

* Is Fork: A boolean value indicating whether the repository is a fork of another repository. In this case, it's false, meaning it is not a fork

* Is Archived: A boolean value indicating whether the repository is archived. Archived repositories are typically read-only and are no longer actively maintained

* Is Template: A boolean value indicating whether the repository is set up as a template

* Default Branch: The name of the default branch

# Question

> "How does the number of "Forks" on GitHub repositories relate to the number of "Stars" and "Watchers"?" 

* Response variable:

`Forks`: The number of times the repository has been forked by other GitHub users

* Explanatory variable:

`Stars`: The number of stars or likes that the repository has received from other GitHub users, indicating its popularity or interest


`Watchers`: The number of GitHub users who are "watching" or monitoring the repository for updates and changes

#### Explanation


The dataset includes information on the number of forks, stars, and watchers of GitHub repositories. This project will aim to explore the relationship between these variables. Specifically, I am going to investigate if the higher number of stars and watchers indicate the higher number of repository being forked.

This project will be focusing on relationship between Forks, Stars and Watchers. Therefore, I will be using **inference** to investigate how the explanatory variables influence the number of repository being forked.

